app/javascript/packs/dashboardDropdowns.js

```diff
@@ -0,0 +1,130 @@
+import { initializeDropdown } from '@utilities/dropdownUtils';
+
+const ARTICLE_ID_REGEX = /\/(\d+)$/;
+const ARTICLE_FORM_KEY_REGEX = /\[(.*)\]/;
+
+function getFormValues(form) {
+  const articleId = form.action.match(ARTICLE_ID_REGEX)[1];
+  const inputs = form.getElementsByTagName('input');
+  const formData = { id: articleId, article: {} };
+
+  for (let i = 0; i < inputs.length; i++) {
+    const input = inputs[i];
+    const name = input.getAttribute('name');
+    const value = input.getAttribute('value');
+
+    const articleFormKeyMatches = name.match(ARTICLE_FORM_KEY_REGEX);
+    if (articleFormKeyMatches) {
+      const key = articleFormKeyMatches[1];
+      formData.article[key] = value;
+    } else {
+      formData[name] = value;
+    }
+  }
+
+  return formData;
+}
+
+function toggleArchived(article, needsArchived) {
+  if (needsArchived === 'true') {
+    article.classList.add('story-archived', 'hidden');
+  } else {
+    article.classList.remove('story-archived');
+  }
+}
+
+function toggleNotifications(submit, action) {
+  if (action === 'Mute Notifications') {
+    submit.setAttribute('value', 'Receive Notifications');
+  } else {
+    submit.setAttribute('value', 'Mute Notifications');
+  }
+}
+
+function onXhrSuccess(form, article, values) {
+  if (values.article.archived) {
+    toggleArchived(article, values.article.archived);
+  } else {
+    const submit = form.querySelector('[type="submit"]');
+    const submitValue = submit.getAttribute('value');
+
+    toggleNotifications(submit, submitValue);
+  }
+}
+
+const handleFormSubmit = (e) => {
+  e.preventDefault();
+  e.stopPropagation();
+
+  const { target: form } = e;
+  const values = getFormValues(form);
+  const data = JSON.stringify(values);
+
+  const formData = new FormData(form);
+  const method = formData.get('_method') || 'post';
+
+  const xhr = new XMLHttpRequest();
```

Comment:
wish we could switch to `window.fetch()` but that's for another day :D

Reply:
It would be some good cleanup and it would probably make for a good first PR. We have it in a bunch of places still.